src/pytorch_lightning/loggers/neptune.py

```diff
@@ -618,18 +619,18 @@ def _dict_paths(cls, d: dict, path_in_build: str = None) -> Generator:
     @property
     def name(self) -> str:
         """Return the experiment name or 'offline-name' when exp is run in offline mode."""
-        return self._run_name
+        return self._run_name  # type: ignore[return-value]
```

Comment:
`_run_name` is annotated with Optional[None] in \_\_init_\_\, and is then set to a string in `._retrieve_run_data`

Reply:
This should already be `str` by the time it's used, shouldn't it? If so, I'm voting for `assert` instead of `# type: ignore`

Reply:
updated in https://github.com/Lightning-AI/lightning/pull/13692/commits/575f150fdb8cd7406d3a04f43a35be1fb1e63618

Reply:
So, it seems this broke couple tests. Something fishy going on, I will debug it next week.

Reply:
would it be more appropriate to use

```python
    @property
    def name(self) -> Optional[str]:
        """Return the experiment name or 'offline-name' when exp is run in offline mode."""
        if self._run_name is not None:
            return self._run_name

    @property
    def version(self) -> Optional[str]:
        """Return the experiment version.

        It's Neptune Run's short_id
        """
        if self._run_short_id is not None:
            return self._run_short_id
```

this passes mypy checks; I can at least push the change so that the other CI/CD checks can be compared across the two commits

Reply:
implemented the above in https://github.com/Lightning-AI/lightning/pull/13692/commits/987337f647be9014f74c2efde6f8efde879b0de2